src/hatch_fancy_pypi_readme/_fragments.py

```diff
@@ -11,52 +11,46 @@
 from pathlib import Path
 from typing import ClassVar, Iterable
 
+from jsonschema import Draft202012Validator, Validator
+
 
 if sys.version_info >= (3, 8):
     from typing import Protocol
 else:
     from typing_extensions import Protocol
-from .exceptions import ConfigurationError
-
-
-def load_fragments(config: list[dict[str, str]]) -> list[Fragment]:
-    """
-    Load all fragments from the fragments config list.
-
-    Raise ConfigurationError on unknown or misconfigured ones.
-    """
-    if not config:
-        raise ConfigurationError(
-            [
-                "tool.hatch.metadata.hooks.fancy-pypi-readme.fragments must "
-                "not be empty."
-            ]
-        )
-
-    frags = []
-    errs = []
-    for frag_cfg in config:
-        for frag in _VALID_FRAGMENTS:
-            if frag.key not in frag_cfg:
-                continue
 
-            try:
-                frags.append(frag.from_config(frag_cfg))
-            except ConfigurationError as e:
-                errs.extend(e.errors)
-
-            break
-        else:
-            errs.append(f"Unknown fragment type {frag_cfg!r}.")
+from .exceptions import ConfigurationError
 
-    if errs:
-        raise ConfigurationError(errs)
 
-    return frags
+TEXT_V = Draft202012Validator(
+    {
+        "type": "object",
+        "properties": {"text": {"type": "string", "pattern": ".+"}},
```

Comment:
All these `"pattern": ".+"` can be `"minLength": 1` if you just want the string to be nonempty.

Reply:
ugh i can't believe i've missed that one